CMakeLists.txt

```
@@ -0,0 +1,63 @@
+cmake_minimum_required(VERSION 3.14)
+
+## Set Project and Version
+file(STRINGS "${CMAKE_CURRENT_SOURCE_DIR}/plugin_name.txt" PLUGIN_NAME)
+file(STRINGS "${CMAKE_CURRENT_SOURCE_DIR}/version.txt" VERSION)
+project("${PLUGIN_NAME}" VERSION "${VERSION}" LANGUAGES CXX)
+
+## Get CMaize
+include(cmake/get_cmaize.cmake)
+
+## Paths ##
+set(PLUGIN_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/src")
```

```suggestion
set(${PROJECT_NAME}_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/src")
```
It's ugly, but this will help avoid variable clashes in case the build system for two plugins are in scope at the same time.